### PR TITLE
Replace or remove update with replace

### PR DESCRIPTION
**Pull request by tkrille**
OSIAM 3.0 removes support for updates via PATCH, so we have to test the
PUT-based API from now on and remove the tests for the PATCH-based one.

Depends on osiam/connector4java#210

### CODE REVIEW

**Comment by tkrille** [comment]
Depends on https://github.com/osiam/connector4java-integration-tests/pull/343

File: src/test/groovy/org/osiam/test/integration/ControllerIT.groovy [hunk 1]
```diff
@@ -319,34 +318,21 @@ class ControllerIT extends AbstractIT {
     def 'OSNG-467: Deactivating a user should revoke his access token'() {
         given: 'active user with valid access token'
         def userId = "cef9452e-00a9-4cec-a086-d171374ffbef"
-        AccessToken serviceAccessToken = OSIAM_CONNECTOR.retrieveAccessToken(Scope.ADMIN)
-        UpdateUser updateUser = new UpdateUser.Builder().updateActive(false).build()
+        def serviceAccessToken = OSIAM_CONNECTOR.retrieveAccessToken(Scope.ADMIN)
+        def user = OSIAM_CONNECTOR.getUser(userId, serviceAccessToken)
+        def updateUser = new User.Builder(user).setActive(false).build()
```
Comment (wallner) [inline src/test/groovy/org/osiam/test/integration/ControllerIT.groovy]: Please move the initialization down into the 'when' block. It makes it easier to understand.

File: src/test/groovy/org/osiam/test/integration/UserGroupMembershipIT.groovy [hunk 2]
```diff
@@ -46,12 +46,12 @@ class UserGroupMembershipIT extends AbstractIT {
         def createdUser = OSIAM_CONNECTOR.createUser(user, OSIAM_CONNECTOR.retrieveAccessToken(Scope.ADMIN))
         def createdGroup = OSIAM_CONNECTOR.createGroup(group, OSIAM_CONNECTOR.retrieveAccessToken(Scope.ADMIN))
 
-        def updateGroup = new UpdateGroup.Builder()
-                .addMember(createdUser.getId())
+        def updateGroup = new Group.Builder(createdGroup)
```
Comment (wallner) [inline src/test/groovy/org/osiam/test/integration/UserGroupMembershipIT.groovy]: I would suggest movingthe initialization into the 'when' block.

Comment (tkrille) [inline src/test/groovy/org/osiam/test/integration/UserGroupMembershipIT.groovy]: Done

File: src/test/groovy/org/osiam/test/integration/ControllerIT.groovy (hunk 1 above)
Comment (tkrille) [inline src/test/groovy/org/osiam/test/integration/ControllerIT.groovy]: Done

**Comment by tkrille** [comment]
Updated

File: src/test/groovy/org/osiam/test/integration/UserGroupMembershipIT.groovy [hunk 3]
```diff
@@ -104,12 +103,12 @@ class UserGroupMembershipIT extends AbstractIT {
 
         def retParentGroup = OSIAM_CONNECTOR.createGroup(parentGroup, accessToken)
 
-        UpdateGroup updateGroup = new UpdateGroup.Builder()
-                .deleteMember(memberGroup1.getId())
+        def updateGroup = new Group.Builder(retParentGroup)
```
Comment (wallner) [inline src/test/groovy/org/osiam/test/integration/UserGroupMembershipIT.groovy]: Don't you thinkthis belongs to the 'when' block too?

File: src/test/groovy/org/osiam/test/integration/UserGroupMembershipIT.groovy [hunk 4]
```diff
@@ -132,12 +131,12 @@ class UserGroupMembershipIT extends AbstractIT {
 
         def parent = OSIAM_CONNECTOR.createGroup(parentGroup, accessToken)
 
-        UpdateGroup updateGroup = new UpdateGroup.Builder()
-                .deleteMembers()
+        def updateGroup = new Group.Builder(parent)
```
Comment (wallner) [inline src/test/groovy/org/osiam/test/integration/UserGroupMembershipIT.groovy]: Please move the initialization to the 'when' block.

**Comment by tkrille** [comment]
Alright, @wallner, I've just replaced PATCH with PUT and did not touch any other things. I don't think it's viable that I improve all tests I've changed now. I will fix your last 2 comments, but that should be the last improvements.

**Comment by tkrille** [comment]
Updated

**Comment by wallner** [comment]
Ok, thank you!

**Pull request merged**